### PR TITLE
get_data_by_key recognizes yaml content

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

### CODE REVIEW

File: osbs/build/config_map_response.py [hunk 1]
```diff
@@ -58,4 +59,6 @@ def get_data_by_key(self, name):
         if data is None or name not in data:
             return {}
 
+        if name.rsplit('.', 1)[-1] == 'yaml':
```
Comment (lcarva) [inline osbs/build/config_map_response.py]: Can we make this `in ('yaml', 'yml')` since both extensions are usually associate with YAML files?

Comment (rcerven) [reply osbs/build/config_map_response.py]: sure

File: tests/mock_jsons/0.5.4/create_config_map.json [hunk 2]
```diff
@@ -2,7 +2,11 @@
     "apiVersion": "v1",
     "data": {
         "special.how": "\"very\"",
-        "special.type": "{\"quark\": \"charm\"}"
+        "special.type": "{\"quark\": \"charm\"}",
+        "config.yaml": "\"version:1\"",
```
Comment (lcarva) [inline tests/mock_jsons/0.5.4/create_config_map.json]: You probably want:

```python
"config.yaml": "version:1",
"config.yml": "version:2",
"config.ymlll": "{\"version\":3}",
"config.json": "{\"version\":4}",
```

File: tests/mock_jsons/0.5.4/create_config_map.json [hunk 3]
```diff
@@ -2,7 +2,11 @@
     "apiVersion": "v1",
     "data": {
         "special.how": "\"very\"",
-        "special.type": "{\"quark\": \"charm\"}"
+        "special.type": "{\"quark\": \"charm\"}",
+        "config.yaml": "version:1",
+        "config.yml": "version:2",
+        "config.ymlll": "\"version:3\"",
```
Comment (lcarva) [inline tests/mock_jsons/0.5.4/create_config_map.json]: You probably want:
```python
"config.ymlll": "{\"version\":3}",
"config.json": "{\"version\":4}",
```

Comment (rcerven) [reply tests/mock_jsons/0.5.4/create_config_map.json]: ok

**Comment by mlangsdorf** [comment]
in the release notes, added "- config_maps can read from json or yaml"
to # API changes

**Pull request merged**